### PR TITLE
[4.4] Fix default parameters for pagination

### PR DESCRIPTION
**Pull request by Fedik**
Pull Request for Issue # .

### Summary of Changes

Added a few missing but required parameters. 
And replaced use of deprecated app->input to modern.

### Testing Instructions
code review

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed
- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

ping @SniperSister

### CODE REVIEW

File: libraries/src/Pagination/Pagination.php [hunk 1]
```diff
@@ -665,21 +665,27 @@ protected function _buildDataObject()
...
-            'tpl'    => 'CMD',
-            'id'     => 'INT',
-            'Itemid' => 'INT',
+            'format'        => 'CMD',
+            'option'        => 'CMD',
+            'controller'    => 'CMD',
+            'view'          => 'CMD',
+            'layout'        => 'CMD',
+            'task'          => 'CMD',
+            'template'      => 'CMD',
```
Comment (SniperSister) [inline libraries/src/Pagination/Pagination.php:674]: Where is "template" used as a query parameter?

Comment (Fedik) [reply libraries/src/Pagination/Pagination.php:674]: Here https://github.com/joomla/joomla-cms/blob/6f2cec29a1d681fb127c40ec60e9493adb0970ab/libraries/src/Application/SiteApplication.php#L483

**Comment by SniperSister** [comment]
I have tested this item :white_check_mark: successfully on 43605d491c7cfab7542dcb52d780aeb18d06f5a0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43954">issues.joomla.org/tracker/joomla-cms/43954</a>.</sub>

File: libraries/src/Pagination/Pagination.php [hunk 2]
```diff
@@ -665,21 +665,27 @@ protected function _buildDataObject()
...
-            'view'   => 'WORD',
-            'layout' => 'WORD',
-            'tpl'    => 'CMD',
-            'id'     => 'INT',
-            'Itemid' => 'INT',
+            'format'        => 'CMD',
+            'option'        => 'CMD',
+            'controller'    => 'CMD',
+            'view'          => 'CMD',
+            'layout'        => 'CMD',
```
Comment (HLeithner) [inline libraries/src/Pagination/Pagination.php]: layout should be 'string'

**Comment by SniperSister** [comment]
Regarding the ID parameter:
If you want to change the INT into a STRING or CMD, make sure to apply the same change to the component cache key:
https://github.com/joomla/joomla-cms/blob/8a868a30a61753d889b70aaf74e3bc738a31be52/libraries/src/Cache/Cache.php#L690

**Comment by SniperSister** [comment]
Same applies to the other suggested changes

**Comment by Hackwar** [comment]
Do we really need to have the cache key parameter be CMD as well? I currently don't see the necessity for that...

**Comment by SniperSister** [comment]
> Do we really need to have the cache key parameter be CMD as well? I currently don't see the necessity for that...

Yes we do, otherwise we re-introduce the issue again.

**Comment by Hackwar** [comment]
But having it be INT is far more restrictive than CMD and we only need the CMD for the router...

File: libraries/src/Pagination/Pagination.php (hunk 2 above)
Comment (Fedik) [reply libraries/src/Pagination/Pagination.php]: yeah, sorry, I missed it,
I tried to check what the cms actualy use for every case.

**Pull request closed without merging**